presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestArrowFlightEchoQueries.java

```diff
@@ -159,6 +171,76 @@ protected QueryRunner createQueryRunner()
+                TimeStampMilliVector timestampVector = new TimeStampMilliVector("timestamp", bufferAllocator);
+                VectorSchemaRoot root = new VectorSchemaRoot(Arrays.asList(intVector, dateVector, timeVector, timestampVector));
+                FlightClient client = createFlightClient(bufferAllocator, serverPort)) {
+            MaterializedResult.Builder expectedBuilder = resultBuilder(getSession(), INTEGER, DATE, TIME, TIMESTAMP);
+
+            intVector.setSafe(0, 0);
+            dateVector.setSafe(0, 0);
+            timeVector.setSafe(0, 0);
+            timestampVector.setSafe(0, 0);
+            expectedBuilder.row(0, LocalDate.parse("1970-01-01"), LocalTime.of(0, 0), LocalDateTime.parse("1970-01-01T00:00:00"));
+
+            intVector.setSafe(1, 1);
+            LocalDate date = LocalDate.of(2024, 1, 1);
+            dateVector.setSafe(1, (int) date.toEpochDay());
+            LocalTime time = LocalTime.of(1, 1, 1);
+            timeVector.setSafe(1, (int) TimeUnit.NANOSECONDS.toMillis(time.toNanoOfDay()));
+            timestampVector.setSafe(1, parseTimestampWithoutTimeZone("2024-01-01 01:01:01"));
+            expectedBuilder.row(1, date, time, LocalDateTime.parse("2024-01-01T01:01:01"));
+
+            intVector.setSafe(2, 2);
+            date = LocalDate.of(2024, 1, 2);
+            dateVector.setSafe(2, (int) date.toEpochDay());
+            time = LocalTime.of(2, 12);
+            timeVector.setSafe(2, (int) TimeUnit.NANOSECONDS.toMillis(time.toNanoOfDay()));
+            timestampVector.setSafe(2, parseTimestampWithoutTimeZone("2024-01-02 12:00:00"));
+            expectedBuilder.row(2, date, time, LocalDateTime.parse("2024-01-02T12:00:00"));
+
+            intVector.setSafe(3, 3);
+            date = LocalDate.of(2112, 12, 31);
+            dateVector.setSafe(3, (int) date.toEpochDay());
+            time = LocalTime.of(23, 58);
+            timeVector.setSafe(3, (int) TimeUnit.NANOSECONDS.toMillis(time.toNanoOfDay()));
+            timestampVector.setSafe(3, parseTimestampWithoutTimeZone("2112-12-31 23:58:00"));
+            expectedBuilder.row(3, date, time, LocalDateTime.parse("2112-12-31T23:58:00"));
+
+            intVector.setSafe(4, 4);
+            date = LocalDate.of(1968, 7, 5);
+            dateVector.setSafe(4, (int) date.toEpochDay());
+            time = LocalTime.of(8, 15, 12, (int) TimeUnit.MILLISECONDS.toNanos(345));
+            timeVector.setSafe(4, (int) TimeUnit.NANOSECONDS.toMillis(time.toNanoOfDay()));
+            timestampVector.setSafe(4, parseTimestampWithoutTimeZone("1968-07-05 08:15:12.345"));
+            expectedBuilder.row(4, date, time, LocalDateTime.parse("1968-07-05T08:15:12.345"));
+
+            root.setRowCount(5);
+
+            String tableName = "datetime";
+            addTableToServer(client, root, tableName);
+
+            MaterializedResult actual = computeActual(sessionStandardTimestamps, format("SELECT * FROM %s", tableName));
```

Comment:
Can you run this with multiple sessions which change the user's timezone to prove that timestamps are displayed independent of the user's timezone?